File: frontend/src/lib/format-sum.ts
/** Format amount in Uzbek so'm (UZS) with spaced thousands: 35 000 so'm */
export function formatSum(amount: number | string | null | undefined): string {
  const n = Math.round(Number(amount ?? 0));
  if (!Number.isFinite(n)) return "0 so'm";
  const spaced = Math.abs(n)
    .toString()
    .replace(/\B(?=(\d{3})+(?!\d))/g, ' ');
  const sign = n < 0 ? '−' : '';
  return `${sign}${spaced} so'm`;
}

File: apps/customer_mobile/lib/core/utils/format_sum.dart
/// Uzbek so'm formatting — matches web `format-sum.ts`.
String formatSum(num? amount) {
  final n = (amount ?? 0).round();
  final abs = n.abs().toString();
  final buffer = StringBuffer();
  for (var i = 0; i < abs.length; i++) {
    if (i > 0 && (abs.length - i) % 3 == 0) buffer.write(' ');
    buffer.write(abs[i]);
  }
  final sign = n < 0 ? '−' : '';
  return "$sign${buffer.toString()} so'm";
}
